092-self-healing-using-cruise-control.md

```diff
@@ -0,0 +1,253 @@
+
+### Strimzi Operations while self-healing is running/fixing an anomaly 
+
+Self-healing is a very robust and resilient feature due to its ability to tackle issues while healing a cluster.
+
+#### What happens when self-healing is running/fixing an anomaly and brokers are rolled
+
+If some anomaly is getting healed , Cruise Control will then try to fix/mitigate the anomaly by moving the partition replicas across the brokers in the Kafka cluster.
+If, during the process of moving partitions replicas among the brokers in the Kafka cluster, some broker that was being used gets deleted or rolled, then Cruise Control would finish self-healing process and log that `self-healing finished successful with error`.
+If the deleted or rolled broker comes back before the periodic anomaly check interval, then self-healing would start again only for the earlier failed broker with a new anomalyId since it was not fixed properly. 
+In case the deleted broker doesn't come back healthy within the periodic anomaly check interval then another new anomaly for the deleted broker will be created and the earlier detected failed broker would also persist but will be updated with a new anomalyId.
+
+#### What happens when self-healing is running/fixing an anomaly and Kafka Rebalance is approved
+
+If self-healing is ongoing and a `KafkaRebalance` resource is posted in the middle of it then the `KafkaRebalance` resource will move to `NotReady` state, stating that there is already a rebalance ongoing. 
+For example:
+```yaml
+status:
+  conditions:
+  - lastTransitionTime: "2025-01-08T08:24:01.314273452Z"
+    message: 'Error for request: my-cluster-cruise-control.myproject.svc:9090/kafkacruisecontrol/rebalance?json=true&dryrun=false&verbose=true&skip_hard_goal_check=false&rebalance_disk=false.
+      Server returned: Error processing POST request ''/rebalance'' due to: ''java.lang.IllegalStateException:
+      Cannot start a new execution while there is an ongoing execution. Please use
+      stop_ongoing_execution=true to stop ongoing execution and start a new one.''.'
+    reason: CruiseControlRestException
+    status: "True"
+    type: NotReady
+  - lastTransitionTime: "2025-01-08T08:24:01.349653295Z"
+    message: 'Wrong annotation value: approve. The rebalance resource supports the
+      following annotations in NotReady state: [refresh]'
+    reason: InvalidAnnotation
+    status: "True"
+    type: Warning
+  observedGeneration: 1
+```
+
+#### What happens when self-healing is running/fixing an anomaly and Kafka Rebalance is applied to ask for a proposal
+
+If self-healing is ongoing and a `KafkaRebalance` resource is applied in the middle of it then Cruise Control will still generate an optimization proposal. The proposal should be based on the cluster shape at that particular timestamp.
+
+## Affected/not affected projects
+
+A new repository named `kubernetes-event-notifier` will be added under the Strimzi organisation.
```

Comment:
There is also the cluster operator to be affected because we need to allow enabling self-healing, or?